audience_manager/api.rst

```diff
@@ -0,0 +1,213 @@
+    </script>
+
+API function
+------------
+
+Loading snippet creates following API function:
+
+.. function:: ppms.am.api(command, ...args)
+
+    JavaScript API interface.
+
+    :param string command: Command name.
+    :param args: Command arguments. Number of arguments and their function depend on command.
+    :returns: Nothing. All commands are prepared to be run asynchronously.
+
+
+Commands
+--------
+Following commands work in context of current browser user profile. Additionally these commands require communication
+with server so they are asynchronous. Callback functions are used to provide response or error information.
+
+List audiences
+``````````````
+Gets list of user audiences.
+
+Code::
+
+    ppms.am.api('getAudiences', onFulfilled, onRejected);
+
+.. function:: onFulfilled(audience_list)
+
+    Function executed on success.
+
+    :param Array<string> audience_list: Array of audience IDs user belongs to. Example::
+
+        ['e8c6e873-955c-4771-9fd5-92c94577e9d9', '756e5920-422f-4d13-b73a-917f696ca288']
+
+.. function:: onRejected(error)
+
+    Function executed on error.
+
+    :param string error: Error code. Example::
+
+        'server_error'
+
+Check audience
+``````````````
+Checks if user is part of specific audience.
+
+Code::
+
+    ppms.am.api('checkAudience', audience_id, onFulfilled, onRejected);
+
+.. data:: audience_id
+
+    ID of checked audience. Example::
+
+        '52073260-5861-4a56-be5e-6628794722ee'
+
+.. function:: onFulfilled(in_audience)
+
+    Function executed on success.
+
+    :param boolean in_audience: True when user is part of audience, false otherwise.
+
+.. function:: onRejected(error)
+
+    Function executed on error.
+
+    :param string error: Error code. Example::
+
+        'server_error'
+
+Get attributes
+``````````````
+Gets user attributes.
+
+.. warning::
+
+    Only white-listed attributes are available this way.
+
+Code::
+
+    ppms.am.api('getAttributes', onFulfilled, onRejected);
+
+.. function:: onFulfilled(attributes)
+
+    Function executed on success.
+
+    :param Object<string,Object<string,string>> attributes: Object containing all white-listed attributes. It's divided
+        on by categories. 'analytics' object contain analytical data about the user (e.g. browser name and version,
+        country, etc.). 'attributes' object contain data from other sources (e.g. Audience Manager Form Tracker, data
+        merged via CSV import etc.). Example::
+
+            {
+                "analytics": {
+                    "browser_name": "chrome",
+                    "country": "us",
+                },
+                "attributes": {
+                    "first_name": "James",
+                    "last_name": "Bond"
+                }
+            }
+
+.. function:: onRejected(error)
+
+    Function executed on error.
+
+    :param string error: Error code. Example::
+
+        'server_error'
+
+Update attributes
+`````````````````
+Updates user attributes.
```

Comment:
Questions that come to my mind when I read this short sentence:
- Do I update only values of existing attributes?
- Can I create a new attribute?
- If I will use attribute name from "Analytics" section (i.e. location), will that be overwritten?

Those descriptions are very short and should give some background so that developers understand AM behavior better. See backend API docs for inspiration: http://am.devtools.piwik.pro/#tag/attributes%2Fpaths%2F~1audiences~1tracker~1attributes~1%2Fpost